ls1intum/Artemis
# [Feature] [Exam-Mode] Automatically assess text and modeling exercises for unsubmitted student exams

### Checklist

### Motivation and Context
Students are required to submit their student exam in the student exam end page. Exercise participation of unsubmitted student exams should not be eligible for assessment. This PR adds a button to the Student Exams page which automatically assessed modeling and text exercises of unsubmitted exams with 0 points.
Additionally, in this PR we also want to assess empty submissions for modeling and text exercises. 

### Description
Client: Add a button to the student exams view.
Server: 
1. Add a result to all text and modeling submissions which are part participations of unsubmitted student exams with 0 points. The result is of type `SEMI_AUTOMATIC` to make sure that these submissions are not eligible for manual assessment in the assessment dashboard. The assessor is the instructor which performs the call. 
2. Add a result to all `empty` latest text and modeling submissions which are part participations of student exams with 0 points. The result is of type `SEMI_AUTOMATIC` to make sure that these submissions are not eligible for manual assessment in the assessment dashboard. The assessor is the instructor which performs the call. 

### Steps for Testing

1. Log in to Artemis
2. Create an exam with all exercise types
3. Register 4 users, start the exam
4. Two users should submit, the other two should not. 
    1. Leave the submissions empty for one user who submits and one who does not the exam.
5. Go to the Student Exams page and click "Assess unsubmitted student exams"
6. The student exams should be assessed.
7. Go to the assessment dashboard and assess the exercises. Only the submissions of the user who submitted the exam and did not leave them empty should be eligible for assessment. The others should **NOT** appear.
8. Set the release date of the results to now. 
9. Review the results as a student and check the feedback. It should say either "Empty Submission" or "You did not submit your exam" for the corresponding submissions


### Test Coverage
With @stefanwaldhauser help we set up a test suite according to the best practices for this component. Currently it only includes tests for the feature implemented in this PR. In a follow-up PR I will improve the test suite to cover the components other functionality.

### Screenshots
![image](https://user-images.githubusercontent.com/15110960/102135913-f4339f00-3e58-11eb-9377-bfc258f15224.png)
![image](https://user-images.githubusercontent.com/15110960/102135963-0c0b2300-3e59-11eb-8e39-7286431a5b6c.png)
![image](https://user-images.githubusercontent.com/15110960/102136274-77ed8b80-3e59-11eb-8bad-bedf797c2ffc.png)
![image](https://user-images.githubusercontent.com/15110960/102253485-a9249500-3f07-11eb-805a-38f74f9a1d4e.png)
![image](https://user-images.githubusercontent.com/15110960/102391144-9a0c1880-3fd5-11eb-9752-23dae9546562.png)
